### PR TITLE
Order supply pack + add search function

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Put all ammo next to their weapon in req
Put explosives (mines, grenades, mortar) in explosives
Put droids in vehicles
Remove attachments from req

## Changelog
:cl:
qol: Req supply back reorganised
qol: Added a search function to req
del: Remove attachment from req
/:cl:

### CODE REVIEW

**Comment by MereWalker** [comment]
Maybe make it a drop-down menu? Click on weapon order. Click on it again to get options for ammo/ ammo types it has. Just my suggestion.

**Comment by BraveMole** [comment]
> Maybe make it a drop-down menu? Click on weapon order. Click on it again to get options for ammo/ ammo types it has. Just my suggestion.

Meh, sometimes you just want to buy ammo and a dropdown would make things slower

File: tgui/packages/tgui/interfaces/Cargo.js [hunk 1]
```diff
@@ -485,58 +483,89 @@ const Category = (props, context) => {
...
+                    <CategoryButton
+                      icon="fast-forward"
+                      disabled={cost > spare_points}
+                      id={entry}
+                      mode="addall" />
+                  </Table.Cell>
+                  <Table.Cell>
+                    <Pack pack={entry} />
+                  </Table.Cell>
+                </Table.Row>
```
Comment (psykzz) [inline tgui/packages/tgui/interfaces/Cargo.js:563]: This should go into it's own component.

File: tgui/packages/tgui/interfaces/Cargo.js [hunk 2]
```diff
@@ -485,58 +483,89 @@ const Category = (props, context) => {
...
+  ] = useLocalState(context, `pack-name-filter`, null);
+
+  const filterSearch = (entry) =>
+    should_filter
+      ? !filter // If we don't have a filter, don't filter
+        ? true // Show everything
+        : supplypackscontents[entry].name
+          ?.toLowerCase()
+          .includes(filter.toLowerCase()) // simple contains search
+      : true;
```
Comment (psykzz) [inline tgui/packages/tgui/interfaces/Cargo.js:502]:
```suggestion
const filterSearch = entry =>
  should_filter && filter
    ? supplypackscontents[entry].name
        ?.toLowerCase()
        .includes(filter.toLowerCase())
    : true;
```

Any reason you couldn't simplify this?

**Pull request merged**